gammapy/makers/map.py

```diff
@@ -75,6 +78,35 @@ def make_counts(geom, observation):
         counts = Map.from_geom(geom)
         counts.fill_events(observation.events)
         return counts
+    
+    @staticmethod
+    def make_counts_rad_max(geom, observation):
+        """Extract the counts using for the ON region size the values in the `RAD_MAX_2D` table"""
+        if observation.rad_max is None:
+            raise TypeError("the IRF for this observation does not include a RAD_MAX_2D table")
+
+        # TODO: move this in a function similar to the make_map_<IRF_COMPONENT>
+        on_center = geom.center_skydir
+        offset = on_center.separation(observation.pointing_radec)
+        rad_max = observation.rad_max.evaluate(
+            offset=offset,
+            energy=geom.axes["energy"].center
+        )
+
+        counts_list = []
+        # create and fill a map per each energy bin, fetch the counts
+        for i, rad in enumerate(rad_max):
+            on_region = CircleSkyRegion(center=on_center, radius=rad)
+            energy_range = geom.axes["energy"].slice(i)
+            on_region_geom = RegionGeom(on_region, axes=[energy_range])
+            counts = Map.from_geom(on_region_geom)
+            counts.fill_events(observation.events)
```

Comment:
Because of the lazy loading the access to `observation.events` will actually re-load the event from disk. To prevent this just introduce a local variable at the top of the method:

```python3
@staticmethod
def make_counts(geom, observation):
    events = observation.events
    ...
```